src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
```

Comment:
Checking for 'factory == null' is not necessary here. The sessionFactory should be configured before the DAO is created, and if it's null, it should fail fast during the construction of the DAO, not during the operations.